website/docs/r/compute_router.html.markdown

```diff
@@ -1,152 +1,153 @@
 
-* `name` - (Required) A unique name for the router, required by GCE. Changing
-    this forces a new router to be created.
-
-* `network` - (Required) The name or resource link to the network this Cloud Router
-    will use to learn and announce routes. Changing this forces a new router to be created.
+* `name` -
+  (Required)
+  Name of the resource. The name must be 1-63 characters long, and
+  comply with RFC1035. Specifically, the name must be 1-63 characters
+  long and match the regular expression `[a-z]([-a-z0-9]*[a-z0-9])?`
+  which means the first character must be a lowercase letter, and all
+  following characters must be a dash, lowercase letter, or digit,
+  except the last character, which cannot be a dash.
+* `network` -
+  (Required)
+  A reference to the network to which this router belongs.
 
-* `bgp` - (Required) BGP information specific to this router.
-    Changing this forces a new router to be created.
-    Structure is documented below.
 
 - - -
 
-* `description` - (Optional) A description of the resource.
-    Changing this forces a new router to be created.
+* `description` -
+  (Optional)
+  An optional description of this resource.
+* `bgp` -
+  (Optional)
+  BGP information specific to this router.  Structure is documented below.
+* `region` -
+  (Optional)
+  Region where the router resides.
+* `project` (Optional) The ID of the project in which the resource belongs.
+    If it is not provided, the provider project is used.
 
-* `project` - (Optional) The ID of the project in which the resource belongs. If it
-    is not provided, the provider project is used.
-    Changing this forces a new router to be created.
+The `bgp` block supports:
+* `asn` -
+  (Required)
+  Local BGP Autonomous System Number (ASN). Must be an RFC6996
+  private ASN, either 16-bit or 32-bit. The value will be fixed for
+  this router resource. All VPN tunnels that link to this router
+  will have the same local ASN.
+* `advertise_mode` -
+  (Optional)
+  User-specified flag to indicate which mode to use for advertisement.
+
+  Valid values of this enum field are: DEFAULT, CUSTOM
+* `advertised_groups` -
+  (Optional)
+  User-specified list of prefix groups to advertise in custom mode.
+  This field can only be populated if advertiseMode is CUSTOM and
+  is advertised to all peers of the router. These groups will be
+  advertised in addition to any specified prefixes. Leave this field
+  blank to advertise no custom groups.
+
+  This enum field has the one valid value: ALL_SUBNETS
+* `advertised_ip_ranges` -
+  (Optional)
+  User-specified list of individual IP ranges to advertise in
+  custom mode. This field can only be populated if advertiseMode
+  is CUSTOM and is advertised to all peers of the router. These IP
+  ranges will be advertised in addition to any specified groups.
+  Leave this field blank to advertise no custom IP ranges.  Structure is documented below.
+        The `advertised_ip_ranges` block supports:
```

Comment:
There are some issues with these autogenerated docs - this tab is one - but I *think* my change will resolve those.  Probably fine to just rebase and regenerate.